R/dash.R

```diff
@@ -211,23 +211,37 @@ Dash <- R6::R6Class(
         #
         # https://cran.r-project.org/doc/FAQ/R-FAQ.html#Others:
         callback_args <- list()
-
+        
         for (input_element in request$body$inputs) {
-          if(is.null(input_element$value))
+          if ("id.index" %in% names(unlist(input_element))) {
+            unlisted_input <- unlist(input_element)
```

Comment:
I'm a bit hesitant to change this part for a couple of reasons. For a normal callback, `request$body$inputs` is already a named list, and unlisting that would make handling regular callbacks more complicated as we then have to append a list of lists to `callback_args` for default callbacks. 

In the case of pattern matching callbacks, each `input_element` is a list of lists. By unlisting it we can access each of the wildcard inputs. I tried to use a vector to subset as you suggested, but while it works for some scenarios, on the initial call of the PMC we run into a subscript out of bounds error, and it might overcomplicate it to account for another condition. 

I'll try to figure out a way to simplify it and improve readability, but I think unlisting with a `for loop` in this case would have to be a necessary evil.

Reply:
OK, trust your judgment (since I do as well). Though one remark I'd make is that the out of bounds error may be related to the current default values for callback inputs, which is `list(NULL)`. In that case, we could probably just ensure these values are handled properly, but don't worry about this for now.

We can always revisit the implementation later if we have an epiphany about making it a little more concise, for the moment it looks OK as-is.